lib/cli-engine.js

```diff
@@ -619,22 +557,16 @@ CLIEngine.prototype = {
         /**
          * Executes the linter on a file defined by the `filename`. Skips
          * unsupported file extensions and any files that are already linted.
-         * @param {string} filename The file to be linted
+         * @param {string} filename The resolved filename of the file to be linted
          * @returns {void}
          */
         function executeOnFile(filename) {
-            var shouldIgnore = ignoredPaths.contains(filename);
-            if (shouldIgnore) {
-                return;
-            }
+            var hashOfConfig;
 
-            filename = fs.realpathSync(filename);
             if (processed[filename]) {
```

Comment:
I'm not sure this is necessary anymore, since you are resolving all directories to files up front anyways.